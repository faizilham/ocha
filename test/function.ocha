fn hello() {
    print("hello");
}

fn world() {
    print("world");
}

fn call_hello_world(is_hello) {
    if (is_hello) {
        hello();
    } else {
        world();
    }

    print("called");
}


call_hello_world(true);             // hello \n called
call_hello_world(false);            // world \n called

fn factorial(n) {
    if (n < 2) {
        return 1;
    } else {
        return factorial(n - 1) * n;
    }
}

print(-factorial(5));               // -120

fn factorial_loop(n) {
    let total = 1;

    while (n >= 2) {
        total = total * n;
        n = n - 1;
    }

    return total;
}

print(factorial_loop(6));                // 720
