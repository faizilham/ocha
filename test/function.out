hello
called
world
called
-120
720
